nixos/tests/clevis.nix

```diff
@@ -0,0 +1,440 @@
+          qemu.options = [
+            "-chardev socket,id=chrtpm,path=/tmp/swtpm-sock"
+            "-tpmdev emulator,id=tpm0,chardev=chrtpm"
+            "-device tpm-tis,tpmdev=tpm0"
+          ];
+        };
+
+        # The test cannot access the network, so any packages we
+        # need must be included in the VM.
+        system.extraDependencies = with pkgs;
+          let
+            python = python3Minimal.override { self = python; includeSiteCustomize = true; };
+            zfsSupport = any (x: x == "zfs") (extraInstallerConfig.boot.supportedFilesystems or [ ]);
+          in
+          [
+            bintools
+            brotli
+            brotli.dev
+            brotli.lib
+            curl
+            desktop-file-utils
+            docbook5
+            docbook_xsl_ns
+            (grub2.override { inherit zfsSupport; })
+            (grub2_efi.override { inherit zfsSupport; })
+            kbd.dev
+            klibc
+            kmod.dev
+            libarchive.dev
+            libxml2.bin
+            libxslt.bin
+            nixos-artwork.wallpapers.simple-dark-gray-bottom
+            ntp
+            perlPackages.ListCompare
+            perlPackages.XMLLibXML
+            python3Minimal
+            (python.withPackages (p: [ p.mistune ]))
+            shared-mime-info
+            sudo
+            texinfo
+            unionfs-fuse
+            xorg.lndir
+          ];
+      };
+    };
+
+    testScript = testScriptFun { inherit createRootPartition extraConfig systemdStage1 fallback preBootCommands; };
+  };
+
+  # The configuration to install to the machine that will use Clevis.
+  makeConfig = { extraConfig, systemdStage1 }:
+    pkgs.writeText "configuration.nix" ''
+      { config, lib, pkgs, ... }:
+
+      {
+        imports = [
+          ./hardware-configuration.nix
+          <nixpkgs/nixos/modules/testing/test-instrumentation.nix>
+        ];
+
+        networking.hostName = "installed";
+
+        boot.loader.grub = {
+          extraConfig = "serial; terminal_output serial";
+          device = "/dev/vda";
+          fsIdentifier = "uuid";
+          configurationLimit = 100;
+        };
+
+
+        boot.kernelParams = [ "console=tty0" ${optionalString (!systemdStage1)  ''"ip=192.168.1.1:::255.255.255.0::eth1:none"''} ];
+
+        boot.initrd = {
+          availableKernelModules = [ "tpm_tis" ];
+          clevis = { enable = true; useTang = true; };
+          ${optionalString (!systemdStage1) "network.enable = true;"}
+          ${optionalString systemdStage1 ''
+            systemd = {
+              enable = true;
+              network.enable = true;
+              network.networks."10-static" = {
+                matchConfig.Name = [ "en*" "eth*" ];
+                DHCP = "no";
+                address = [ "192.168.1.1/24" ];
+                linkConfig.RequiredForOnline = true;
+                networkConfig.IPv6PrivacyExtensions = "kernel";
+              };
+            };
+          ''}
```

Comment:
Any reason why you have explicit configuration for systemd here? I'm pretty sure networkd will generate network files for the `ip=` cmdline param. So I'm pretty sure you can just do `boot.initrd.network.enable = true;` in either case.

Reply:
Indeed, it works and it's cleaner, thanks